### PR TITLE
Source record linkage

### PR DESCRIPTION
**Pull request by atz**
- [x] TODO: add some tests

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -217,11 +217,15 @@
     t.datetime "updated_at",                          null: false
     t.string   "doi",                limit: 255
     t.integer  "pmid",               limit: 4
+    t.integer  "publication_id",     limit: 4
   end
 
   add_index "web_of_science_source_records", ["doi"], name: "web_of_science_doi_index", using: :btree
   add_index "web_of_science_source_records", ["pmid"], name: "web_of_science_pmid_index", using: :btree
-  add_index "web_of_science_source_records", ["uid"], name: "web_of_science_uid_index", using: :btree
+  add_index "web_of_science_source_records", ["publication_id"], name: "index_web_of_science_source_records_on_publication_id", unique: true, using: :btree
```
[the remark this answers is not in the document]
Comment (atz) [reply db/schema.rb]: Yes, that is controlled by the field declaration, which does not have `null: false`.  It is nullable.

[the remark this answers is not in the document]
Comment (atz) [reply db/schema.rb]: This is important, because the source record is always created before the publication, so it will always start as null.

**Comment by peetucket** [comment]
Just to double check, this is deployable to -dev now

**Comment by atz** [comment]
Right, and that would be a good test of the migrations.

**Pull request merged**